### ISSUE TITLE
Error with latest version of Pandas

### ISSUE THREAD

**Issue by a-dpq** [gerrymanoim/exchange_calendars#40]
I recently updated my packages, and now when calling `from exchange_calendars import get_calendar`, I get the following error:

```
Traceback (most recent call last):
  File "test.py", line 1, in <module>
    from exchange_calendars import get_calendar
  File "/opt/miniconda3/lib/python3.7/site-packages/exchange_calendars/__init__.py", line 16, in <module>
    from .calendar_utils import (
  File "/opt/miniconda3/lib/python3.7/site-packages/exchange_calendars/calendar_utils.py", line 3, in <module>
    from .always_open import AlwaysOpenCalendar
  File "/opt/miniconda3/lib/python3.7/site-packages/exchange_calendars/always_open.py", line 5, in <module>
    from .exchange_calendar import ExchangeCalendar
  File "/opt/miniconda3/lib/python3.7/site-packages/exchange_calendars/exchange_calendar.py", line 27, in <module>
    from .calendar_helpers import (
  File "/opt/miniconda3/lib/python3.7/site-packages/exchange_calendars/calendar_helpers.py", line 6, in <module>
    NP_NAT = np.array([pd.NaT], dtype=np.int64)[0]
TypeError: int() argument must be a string, a bytes-like object or a number, not 'NaTType'

```

This occurs both on macOS with Python 3.7 and Ubuntu with Python 3.8.

**Comment by a-dpq** [comment]
The issue seems to have been introduced with the latest version of Pandas (1.3.0). After downgrading to the previous version (1.2.5), it no longer occurs.

**Comment by drpalmer** [comment]
With the upgrade to Numpy 1.21.0 and Pandas 1.3.0 I've experienced "TypeError: int() argument must be a string, a bytes-like object or a number, not 'NaTType'"

The offending line of code is line 6 in calendar_helpers: NP_NAT = np.array([pd.NaT], dtype=np.int64)[0]

I've changed my local copy to : NP_NAT = np.array([pd.NaT.asm8.view('i8')], dtype=np.int64)[0]

Recompiled and it seems to work fine.

**Comment by maread99** [comment]
Another issue following this latest pandas release is that the ExchangeCalendar constructor is now raising a FutureWarning.

The cause is within the creation of the schedule `DataFrame`, due to conversion of DatetimeIndex values with dtype `'datetime64[ns, UTC]'` (self._opens etc.) to `'datetime64[ns]'`.

```
        self.schedule = DataFrame(
            index=_all_days,
            data=OrderedDict(
                [
                    ("market_open", self._opens),
                    ("break_start", self._break_starts),
                    ("break_end", self._break_ends),
                    ("market_close", self._closes),
                ]
            ),
            dtype="datetime64[ns]",
        )
```

One option would be to change the target dtype to "datetime64[ns, UTC]" thereby defining the schedule columns in terms of UTC. This additional context would be meaningful although I suspect it would open a can of worms with respect to changes it would necessitate within `exchange_calendars` and its clients. For now I'll add a fix to PR #41 that retains the existing behaviour.
